### PR TITLE
fix: rearrange provider tx param error code checks

### LINKED ISSUE #1349: GAS_PRICE_TOO_LOW vs GAS_LIMIT_TOO_LOW - Incorrect error being thrown
https://github.com/FuelLabs/fuels-ts/blob/f6fb12b7fbc5c4ef4ed0b7471ae6749468b84a96/packages/providers/src/provider.ts#L479C4-L489C6

The error being thrown is switched upside down here

### PR DESCRIPTION
**Pull request by danielbate**
Closes #1349

### CODE REVIEW

**Review by arboleya — APPROVED** [review]
How come you added a new test, and the overall coverage decreased?

I'm assuming this is an inaccurate report from the CI action.

File: packages/providers/test/provider.test.ts [hunk 1]
```diff
@@ -830,4 +830,62 @@ describe('Provider', () => {
...
+        code: ErrorCode.GAS_LIMIT_TOO_LOW,
+        message: `Gas limit '${gasLimit}' is lower than the required: '${gasUsed}'.`,
+      }
+    );
+
+    expect(txCostSpy).toHaveBeenCalled();
+    expect(estimateTxSpy).toHaveBeenCalled();
+  });
+
+  it.only('throws when gas price is lower than min tx gas price', async () => {
```
Comment (arboleya) [inline packages/providers/test/provider.test.ts:863]:
```suggestion
  it('throws when gas price is lower than min tx gas price', async () => {
```

**Review by arboleya — CHANGES_REQUESTED** [review]
I found the reason for the coverage decrease.

File: packages/providers/test/provider.test.ts (hunk 1 above)
Comment (danielbate) [reply packages/providers/test/provider.test.ts:863]: Doh

**Review by arboleya — CHANGES_REQUESTED** [review]

**Review by arboleya — APPROVED** [review]

**Review by digorithm — APPROVED** [review]

**Pull request merged**